src/Microsoft.AspNet.Mvc.Core/ActionResults/ContentResult.cs

```diff
@@ -26,17 +31,26 @@ public class ContentResult : ActionResult
         {
             var response = context.HttpContext.Response;
 
-            MediaTypeHeaderValue contentTypeHeader;
-            if (string.IsNullOrEmpty(ContentType))
+            var contentTypeHeader = ContentType;
+            Encoding encoding;
+            if (contentTypeHeader == null)
             {
-                contentTypeHeader = new MediaTypeHeaderValue("text/plain");
+                contentTypeHeader = DefaultContentType;
+                encoding = DefaultContentType.Encoding;
             }
             else
             {
-                contentTypeHeader = new MediaTypeHeaderValue(ContentType);
+                if (contentTypeHeader.Encoding == null)
+                {
+                    // 1. Do not modify the user supplied content type
+                    // 2. Parse here to handle parameters apart from charset
+                    contentTypeHeader = MediaTypeHeaderValue.Parse(contentTypeHeader.ToString());
```

Comment:
we do not want to modify the instance of MediaTypeHeaderValue value that the user supplied to ContentType property and also there could be additional parameters...test: https://github.com/aspnet/Mvc/pull/2450/files#diff-3be60edcf4b71964bc2ec1b25a0984fbR99